cpp/src/plasma/client.cc

```diff
@@ -257,7 +278,7 @@ class PlasmaClient::Impl : public std::enable_shared_from_this<PlasmaClient::Imp
   /// Table of dlmalloc buffer files that have been memory mapped so far. This
   /// is a hash table mapping a file descriptor to a struct containing the
   /// address of the corresponding memory-mapped file.
-  std::unordered_map<int, ClientMmapTableEntry> mmap_table_;
+  std::unordered_map<int, std::unique_ptr<ClientMmapTableEntry>> mmap_table_;
```

Comment:
Why `unique_ptr`? If you do `std::unordered_map<int, ClientMmapTableEntry>`, then when you delete an entry from `mmap_table_`, its destructor will run.

Reply:
Oh do you need that in order to compile when using `ARROW_DISALLOW_COPY_AND_ASSIGN`?

Reply:
Yes, without the unique_ptr it would not be correct (upon rehashing). The constructor will run with unique_ptr as well.